tests/test_requests.py

```diff
@@ -2127,6 +2133,7 @@ def test_preparing_url(self, url, expected):
             b"http://*",
             u"http://*.google.com",
             u"http://*",
+            u"http://☃.net/"
         )
     )
     def test_preparing_bad_url(self, url):
```

Comment:
Hrmm, so I'm assuming the urls in this `test_preparing_bad_url` are to show we prevent wildcards? If we fallback to simply checking for ASCII on failed IDNA encoding, these all pass now. Is there a discrete subset of ASCII URLs that we still want/need to catch?